pyfarm/agent/http/core/client.py

```diff
@@ -304,35 +323,56 @@ def request(method, url, **kwargs):
         The class to use to unpack the internal response.  This is mainly
         used by the unittests but could be used elsewhere to add some
         custom behavior to the unpack process for the incoming response.
+
+    :raises NotImplementedError:
+        Raised whenever a request is made of this function that we can't
+        implement such as an invalid http scheme, request method or a problem
+        constructing data to an api.
     """
-    # check assumptions for arguments
-    assert method in ("HEAD", "GET", "POST", "PUT", "PATCH", "DELETE")
-    assert isinstance(url, STRING_TYPES) and url
+    assert isinstance(url, STRING_TYPES)
+
+    # Make sure it's a method we support.
+    if method not in HTTP_METHODS:
+        raise NotImplementedError(
+            "This function only supports these http "
+            "methods: %s" % ", ".join(HTTP_METHODS))
+
+    # We only support http[s]
+    parsed_url = urlparse(url)
+    if parsed_url.scheme not in HTTP_SCHEMES:
+      raise NotImplementedError("Only http or https is supported.")
+
+    # When using HTTPs there are some specific modules that are
+    # required but not specified as required in the setup.py file.
+    if parsed_url.scheme == "https":
+        if PyOpenSSL is NotImplemented and ssl is NotImplemented:
+            raise NotImplementedError(
+                "HTTPs requires the ssl or PyOpenSSL modules")
```

Comment:
The error treq *used* to raise would tell you nothing about what the real problem was and you'd have to investigate because it was similar to other errors.  That's not the case anymore however it seems, I just checked and now we get this:
```
  File "twisted/web/client.py", line 1560, in request
    endpoint = self._getEndpoint(parsedURI)
  File "twisted/web/client.py", line 1544, in _getEndpoint
    return self._endpointFactory.endpointForURI(uri)
  File "twisted/web/client.py", line 1420, in endpointForURI
    tlsPolicy = self._policyForHTTPS.creatorForNetloc(uri.host, uri.port)
  File "twisted/web/client.py", line 797, in raiseNotImplemented
    raise NotImplementedError("SSL support unavailable")
NotImplementedError: SSL support unavailable
```

So I'll remove this check (sorry didn't do this before, hadn't updated my treq client in a while)